integration-tests/smoke/ccip_test.go

```diff
@@ -168,105 +151,233 @@ func TestTokenTransfer(t *testing.T) {
+			tokenAmounts: []router.ClientEVMTokenAmount{
+				{
+					Token:  srcToken1.Address(),
+					Amount: big.NewInt(1e18),
+				},
+				{
+					Token:  srcToken2.Address(),
+					Amount: big.NewInt(2e18),
+				},
+			},
+			receiver: e.Chains[tenv.FeedChainSel].DeployerKey.From,
+			data:     []byte(""),
+		},
+		{
+			name:     "Send N tokens to contract",
+			srcChain: tenv.HomeChainSel,
+			dstChain: tenv.FeedChainSel,
+			tokenAmounts: []router.ClientEVMTokenAmount{
+				{
+					Token:  srcToken1.Address(),
+					Amount: big.NewInt(1e18),
+				},
+				{
+					Token:  srcToken2.Address(),
+					Amount: big.NewInt(2e18),
+				},
+				{
+					Token:  srcToken1.Address(),
+					Amount: big.NewInt(3e18),
+				},
+			},
+			receiver: state.Chains[tenv.FeedChainSel].Receiver.Address(),
+			data:     []byte(""),
+		},
+	}
 
-	tx, err = dstToken.Mint(
-		e.Chains[tenv.FeedChainSel].DeployerKey,
-		e.Chains[tenv.FeedChainSel].DeployerKey.From,
-		new(big.Int).Mul(twoCoins, big.NewInt(10)),
-	)
-	require.NoError(t, err)
-	_, err = e.Chains[tenv.FeedChainSel].Confirm(tx)
-	require.NoError(t, err)
+	for _, scenario := range scenarios {
+		scenario := scenario
+		t.Run(scenario.name, func(t *testing.T) {
 
-	tx, err = srcToken.Approve(e.Chains[tenv.HomeChainSel].DeployerKey, state.Chains[tenv.HomeChainSel].Router.Address(), twoCoins)
-	require.NoError(t, err)
-	_, err = e.Chains[tenv.HomeChainSel].Confirm(tx)
-	require.NoError(t, err)
-	tx, err = dstToken.Approve(e.Chains[tenv.FeedChainSel].DeployerKey, state.Chains[tenv.FeedChainSel].Router.Address(), twoCoins)
-	require.NoError(t, err)
-	_, err = e.Chains[tenv.FeedChainSel].Confirm(tx)
-	require.NoError(t, err)
+			startBlocks := make(map[uint64]*uint64)
+			expectedSeqNum := make(map[uint64]uint64)
 
-	tokens := map[uint64][]router.ClientEVMTokenAmount{
-		tenv.HomeChainSel: {{
-			Token:  srcToken.Address(),
-			Amount: twoCoins,
-		}},
-		tenv.FeedChainSel: {{
-			Token:  dstToken.Address(),
-			Amount: twoCoins,
-		}},
-	}
-
-	for src := range e.Chains {
-		for dest, destChain := range e.Chains {
-			if src == dest {
-				continue
-			}
+			destChain := e.Chains[scenario.dstChain]
 			latesthdr, err := destChain.Client.HeaderByNumber(testcontext.Get(t), nil)
 			require.NoError(t, err)
 			block := latesthdr.Number.Uint64()
-			startBlocks[dest] = &block
+			startBlocks[scenario.dstChain] = &block
+
+			// Fetch initial balances and aggregate total amounts
+			initialBalances := make(map[common.Address]*big.Int)
+			totalAmountsTransferred := make(map[common.Address]*big.Int)
+
+			for _, tokenAmount := range scenario.tokenAmounts {
+				dstToken := getDestinationToken(scenario, tenv, srcToken1, dstToken1, srcToken2, dstToken2, tokenAmount.Token)
+				require.NotNil(t, dstToken, "Destination token not found")
+
+				// Fetch initial balance
+				if _, exists := initialBalances[dstToken.Address()]; !exists {
+					balance, err := dstToken.BalanceOf(nil, scenario.receiver)
+					require.NoError(t, err)
```

Comment:
Similar story already for USDC, if worth the effort / makes sense we can dedup this code